### PR TITLE
Removing old inefficient set_sidebar #8803

### LINKED ISSUE #8803: Stop using old inefficient `set_sidebar` method
We used to use `set_sidebar` a lot because it was a standard way to show sidebars across the site. But that old style sidebar is essentially completely gone, and we should remove it from any remaining pages. The code is very inefficient and it's causing 25+ second load times. 

The function itself is here, and you can see why it's so inefficient! it's super long and just makes tons of queries for stuff we don't use anymore, like wikis, etc etc:

https://github.com/publiclab/plots2/blob/cbb807ba8e2302f09dafc0060475aa118e34c2c6/app/controllers/application_controller.rb#L25-L67

Here's all the instances we should carefully remove and test:

https://github.com/publiclab/plots2/search?q=set_sidebar

This could be broken out into sub-tasks or sub-issues! We'd love help with it!

### PR DESCRIPTION
**Pull request by gauravahlawat81**
Fixes #8803 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

This is a PR in progress, I have basically just commented all the places where `set_sidebar was` used and I have also tested for changes. As far as I could test, everything was working fine and nothing was broken. I am open to more ideas and suggestions on how I might be able to solve this in a better way.

### CODE REVIEW

**Comment by gauravahlawat81** [comment]
@cesswairimu Could you please test it if this is indeed the correct solution and nothing was broken. I'm all ears for improvement. Thanks !

**Comment by noi5e** [comment]
@gauravahlawat81 The code looks good so far. 👍  

If you scroll down, it looks like the codeclimate check isn't passing. All 3 checks need to pass before someone can merge your pull request.
Screen Shot 2021-01-03 at 6 46 10 PM

If you click on 'Details' next to codeclimate, there will be a write-up on why the changes aren't passing:
Screen Shot 2021-01-03 at 6 48 47 PM

Okay, that actually sounds like an easy fix. It sounds like there were too many blank lines in between method definitions when you deleted `set_sidebar`. If you reduce the number of blank lines, I'm thinking this check will pass.

Comment here if you need help making the changes and making another commit to this PR! 🤗

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -22,49 +22,6 @@ def set_raven_context
     Raven.extra_context(params: params.to_unsafe_h, url: request.url)
   end
```
Comment (noi5e) [inline app/controllers/application_controller.rb:24]: Here's blank line #1.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -22,49 +22,6 @@ def set_raven_context
...
-               end
-
-      @wikis = Node.order('changed DESC')
-        .joins(:revision)
-        .where('node_revisions.status = 1 AND node.status = 1 AND type = "page"')
-        .limit(10)
-        .group('node_revisions.nid')
-        .order('node_revisions.timestamp DESC')
-    end
-  end
```
Comment (noi5e) [inline app/controllers/application_controller.rb:25]: Here's blank line #2.

**Comment by gauravahlawat81** [comment]
@noi5e Thanks for your help, I was successfully able to pass all the tests.

**Comment by gauravahlawat81** [comment]
@jywarren Can you review this PR ?

**Comment by cesswairimu** [comment]
This is really great @gauravahlawat81, let me check if we need to remove some code in the corresponding templates

**Comment by cesswairimu** [comment]
I think we might need to remove the instances where we are rendering these templates `<%= render partial: "sidebar/` but tagging @jywarren  to confirm.

**Comment by jywarren** [comment]
Confirming this - i think that's right we need to remove the views, but... let's first find them and review them - maybe in a new issue? It seems like we aren't breaking any tests by doing this PR so no reason to hold it up for that extra views task, right?

Thanks, everyone!!!!

**Comment by jywarren** [comment]
I see this is still marked WIP but @gauravahlawat81 and @cesswairimu do you think it's complete and ready to merge then?

**Comment by gauravahlawat81** [comment]
> I see this is still marked WIP but @gauravahlawat81 and @cesswairimu do you think it's complete and ready to merge then?

Hey, I just wanted to confirm that whether the changes I have made are good enough or not. I wasn't so sure of I should comment or remove the old instances where `set_sidebar` was being used. So I have made all the changes from my side, and once you finish reviewing it, I think it can be merged!

**Comment by cesswairimu** [comment]
Great 🎉 Thanks

**Review by cesswairimu — APPROVED** [review]

**Comment by cesswairimu** [comment]
Thanks everyone

**Comment by gauravahlawat81** [comment]
Thank you @cesswairimu and @jywarren . My 2nd PR merged, I am loving it so far, contributing to such an awesome and active community !!

**Comment by cesswairimu** [comment]
Thanks for all the great work you are doing @gauravahlawat81 ...I will create a follow up to remove the templates

**Pull request merged**